### PR TITLE
CIMIT Stubs: Add DNS name for Stub API GW

### PR DESCRIPTION
**Pull request by TalatAydinGDS**
### What changed

CIMIT Stubs: Add DNS name for Stub API GW

### Why did it change

 Add DNS name for Stub API GW

### Issue tracking

- [PYIC-3343](https://govukverify.atlassian.net/browse/PYIC-3343)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-3343]: https://govukverify.atlassian.net/browse/PYIC-3343?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TalatAydinGDS** [comment]
deployed and tested with Postman from my dev environment. Please find some Postman scripts here [PYIC-3343]

[PYIC-3343]: https://govukverify.atlassian.net/browse/PYIC-3343

File: di-ipv-cimit-stub/deploy/template.yaml [hunk 1]
```diff
@@ -373,6 +380,68 @@ Resources:
...
+            - !Sub "cimit-${Environment}.core.dev.stubs.account.gov.uk"
+            - !If [ IsNonProd, !Sub "cimit.${Environment}.stubs.account.gov.uk", cimit.stubs.account.gov.uk ]
+          HostedZoneId: !If
+            - IsCoreDev
+            - !ImportValue DevPublicHostedZoneId
+            - !If [ IsNonProd, !ImportValue BuildPublicHostedZoneId, !ImportValue RootPublicHostedZoneId ]
+      ValidationMethod: DNS
+
+  # api domain entries / mapping
+  CimitStubRestApiDomain:
```
Comment (TobySaundersGDS) [inline di-ipv-cimit-stub/deploy/template.yaml:403]: We are not using apigateway v2 - this uses a rest API gateway so you'll need something likethis instead:
```
 # api domain entries / mapping
  RestApiGatewayDomain:
    Type: AWS::ApiGateway::DomainName
    Condition: IsNewDevelopment
    Properties:
      DomainName: !Sub "rest.${Environment}.dev.identity.account.gov.uk"
      RegionalCertificateArn: !Ref RestSSLCert
      EndpointConfiguration:
        Types:
          - REGIONAL

  RestApiBasePathMapping:
    Type: AWS::ApiGateway::BasePathMapping
    Condition: IsNewDevelopment
    Properties:
      DomainName: !Ref RestApiGatewayDomain
      RestApiId: !Ref RestApiGateway
      Stage: !Ref ApiGatewayStage
```

**Review by TobySaundersGDS — CHANGES_REQUESTED** [review]
Can you put in the rest API version for the domain name and mapping?

File: di-ipv-cimit-stub/deploy/template.yaml (hunk 1 above)
Comment (TalatAydinGDS) [reply di-ipv-cimit-stub/deploy/template.yaml:403]: Thank you for the feedback. I have made the changes and retest again with Postman. It's working with version changes.

**Review by TobySaundersGDS — APPROVED** [review]

**Pull request merged**